internal/analyzers/prometheus.go

```diff
@@ -0,0 +1,295 @@
+		LabelSelector: "prometheus=prometheus",
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list RoleBindings: %w", err)
+	}
+	
+	rbacHelper := &rbachelpers.RBACHelper{}
+
+	if !rbacHelper.IsServiceAccountBoundToRoleBindingList(cRb, prometheus.Spec.ServiceAccountName) {
+		return fmt.Errorf("ServiceAccount %s is not bound to any RoleBindings", prometheus.Spec.ServiceAccountName)
+	}
+
+	for _, crb := range cRb.Items {
+		cr, err := clientSets.KClient.RbacV1().ClusterRoles().Get(ctx, crb.RoleRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get ClusterRole %s", crb.RoleRef.Name)
+		}
+
+		err = checkClusterRoleRules(crb, cr)
+		if err != nil {
+			return err
+		}
+	}
+
+	namespaceSelectors := map[string]interface{}{
+		"PodMonitorNamespaceSelector":     prometheus.Spec.PodMonitorNamespaceSelector,
+		"ServiceMonitorNamespaceSelector": prometheus.Spec.ServiceMonitorNamespaceSelector,
+		"ScrapeConfigNamespaceSelector":   prometheus.Spec.ScrapeConfigNamespaceSelector,
+		"ProbeNamespaceSelector":          prometheus.Spec.ProbeNamespaceSelector,
+	}
+
+	namespaceSelectorStatus, nilNamespaceSelectors := checkPrometheusNamespaceSelectorsStatus(namespaceSelectors)
+
+	if namespaceSelectorStatus {
+		slog.Info("Prometheus is compliant, no issues found", "name", name, "namespace", namespace)
+		return nil
+	}
+
+	if !namespaceSelectorStatus {
+		if len(nilNamespaceSelectors) > 0 {
+			checkSelectorsCreated, selectorLists := checkServiceDiscovery(ctx, clientSets, namespace)
+			if !checkSelectorsCreated && selectorLists == nil {
+				return fmt.Errorf("No Service Selectors are created yet")
+			}
+			if len(nilNamespaceSelectors) == len(selectorLists) {
+				fmt.Println("All namespace selectors are either empty or properly defined.")
+			}
+		}
+	}
+
+	serviceSelectors := map[string]interface{}{
+		"ServiceMonitorSelector": prometheus.Spec.ServiceMonitorSelector,
+		"PodMonitorSelector":     prometheus.Spec.PodMonitorSelector,
+		"ProbeSelector":          prometheus.Spec.ProbeSelector,
+		"ScrapeConfigSelector":   prometheus.Spec.ScrapeConfigSelector,
+	}
+
+	serviceSelectorStatus, emptyServiceSelectors := checkPrometheusServiceSelectorsStatus(serviceSelectors)
+
+	if !serviceSelectorStatus {
+		return fmt.Errorf("No Service Selectors are defined, the Prometheus matches no objects, configuration is unmanaged")
+	}
+
+	if serviceSelectorStatus && emptyServiceSelectors!= nil {
+		checkServiceSelectorsCreated, svcselectorsLists := checkServiceDiscovery(ctx, clientSets, namespace)
+		if !checkServiceSelectorsCreated {
+			return fmt.Errorf("No Service Selectors are created yet")
+		}
+		if len(emptyServiceSelectors) == len(svcselectorsLists) {
+			fmt.Println("Prometheus matches all objects.")
+		}
+	}
+
+	slog.Info("Prometheus is compliant, no issues found", "name", name, "namespace", namespace)
+	return nil
+}
+
+func checkClusterRoleRules(crb v1.ClusterRoleBinding, cr *v1.ClusterRole) error {
```

Comment:
Maybe something like that:

```
import (
	"fmt"
	"strings"
)

func checkClusterRoleRules(crb v1.ClusterRoleBinding, cr *v1.ClusterRole) error {
	var errs []string // Collect errors in a slice
	verbsToCheck := []string{"get", "list", "watch"}

	for _, rule := range cr.Rules {
		// Check for configmaps resource
		for _, resource := range rule.Resources {
			if resource == "configmaps" {
				hasGet := false
				for _, verb := range rule.Verbs {
					if verb == "get" {
						hasGet = true
						break
					}
				}
				if !hasGet {
					errs = append(errs, fmt.Sprintf("ClusterRole %s does not include 'configmaps' with 'get' in its verbs", crb.RoleRef.Name))
				}
			}
		}

		// Check necessary verbs for API groups
		if len(rule.APIGroups) > 0 {
			missingVerbs := []string{}
			for _, requiredVerb := range verbsToCheck {
				found := false
				for _, verb := range rule.Verbs {
					if verb == requiredVerb {
						found = true
						break
					}
				}
				if !found {
					missingVerbs = append(missingVerbs, requiredVerb)
				}
			}
			if len(missingVerbs) > 0 {
				errs = append(errs, fmt.Sprintf("ClusterRole %s is missing necessary verbs for APIGroups: %v", crb.RoleRef.Name, missingVerbs))
			}
		}

		// Check NonResourceURLs for metrics
		for _, nonResource := range rule.NonResourceURLs {
			if nonResource == "/metrics" {
				hasGet := false
				for _, verb := range rule.Verbs {
					if verb == "get" {
						hasGet = true
						break
					}
				}
				if !hasGet {
					errs = append(errs, fmt.Sprintf("ClusterRole %s does not include 'get' verb for NonResourceURL '/metrics'", crb.RoleRef.Name))
				}
			}
		}
	}

	// If there are any errors, return them as a single error
	if len(errs) > 0 {
		return fmt.Errorf("Multiple errors found:\n%s", strings.Join(errs, "\n"))
	}

	return nil
}
```

Reply:
This as is doesn't quite work as is going over and giving error for missing verbs, but I see what you mean in terms of getting a single error, will adapt according to that

```
- apiGroups:
  - ""
  resources:
  - configmaps
  verbs:
  - get
```

Reply:
Changed based on your suggestion, pls review